### PR TITLE
Feature/84 console canvas

### PR DESCRIPTION
**Pull request by vladimirchr**
PBI #84

### CODE REVIEW

File: HW3/Vladimir_Chirushin/ConsoleCanvas/ConsoleDrawer.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    public class ConsoleDrawer : IConsoleDrawer
+    {
+        private readonly IDrawManager drawManager;
+        private readonly IPhraseProvider phraseProvider;
+        private readonly IObjectDrawer canvasDrawer;
+        private readonly IObjectDrawer dotDrawer;
+        private readonly IObjectDrawer verticalLineDrawer;
+        private readonly IObjectDrawer horizontalLineDrawer;
+        private readonly IObjectDrawer gooseDrawer;
```
Comment (spore9) [inline HW3/Vladimir_Chirushin/ConsoleCanvas/ConsoleDrawer.cs]: I am not sure, but why not unite what similar objects into one class or at least array or some collection.

File: HW3/Vladimir_Chirushin/ConsoleCanvas/goose.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+░░░░▐░░░░░▐░░░░░░░ 
+░░░░▌░░░░░▐▄▄░░░░░ 
+░░░░▌░░░░▄▀▒▒▀▀▀▀▄ 
+░░░▐░░░░▐▒▒▒▒▒▒▒▒▀▀▄ 
+░░░▐░░░░▐▄▒▒▒▒▒▒▒▒▒▒▀▄ 
+░░░░▀▄░░░░▀▄▒▒▒▒▒▒▒▒▒▒▀▄ 
+░░░░░░▀▄▄▄▄▄█▄▄▄▄▄▄▄▄▄▄▄▀▄ 
+░░░░░░░░░░░▌▌░▌▌░░░░░░░░░░ 
+░░░░░░░░░░░▌▌░▌▌░░░░░░░░░░
+░░░░░░░░░▄▄▌▌▄▌▌░░░░░░░░░░
```
Comment (spore9) [inline HW3/Vladimir_Chirushin/ConsoleCanvas/goose.txt]: Cool picture)

File: HW3/Vladimir_Chirushin/ConsoleCanvas/ConsoleDrawer.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            DrawDelegate gooseDelegate = new DrawDelegate(gooseDrawer.DrawObject);
+
+            do
+            {
+                consoleKeyPressed = keyboardManager.ReadKey();
+
+                switch (consoleKeyPressed.Key)
+                {
+                    case ConsoleKey.D1:
+                        drawingDelegates += canvasDelegate;
```
Comment (vsatkov) [inline HW3/Vladimir_Chirushin/ConsoleCanvas/ConsoleDrawer.cs]: You should just use 'canvasDrawer.DrawObject' here, it have same signature.
No reasons to extra delegates.

File: HW3/Vladimir_Chirushin/ConsoleCanvas/ConsoleDrawer.cs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+            DrawDelegate horizontalLineDelegate = new DrawDelegate(horizontalLineDrawer.DrawObject);
+            DrawDelegate gooseDelegate = new DrawDelegate(gooseDrawer.DrawObject);
+
+            do
+            {
+                consoleKeyPressed = keyboardManager.ReadKey();
+
+                switch (consoleKeyPressed.Key)
+                {
+                    case ConsoleKey.D1:
```
Comment (vsatkov) [inline HW3/Vladimir_Chirushin/ConsoleCanvas/ConsoleDrawer.cs]: You used 'ConsoleKey.D1' here but I got not board due to using numpad 1. Be careful with such things.

File: HW3/Vladimir_Chirushin/ConsoleCanvas/goose.txt (hunk 2 above)
Comment (vsatkov) [reply HW3/Vladimir_Chirushin/ConsoleCanvas/goose.txt]: Yeap, the goose is cool!

**Pull request merged**